pkg/network/tracer/tracer_test.go

```diff
@@ -960,47 +960,51 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
 	queryMsg.SetQuestion(dns.Fqdn(domain), dns.TypeA)
 	queryMsg.RecursionDesired = true
 
-	var dnsClientAddr *net.UDPAddr
-	require.Eventually(t, func() bool {
+	// we place the entire test in the eventually loop since failed DNS requests can cause "WaitForDomain" to return
+	// true before DNS data is actually available for the successful connection
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
 		dnsClient := new(dns.Client)
 		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-		require.NoError(t, err)
-		dnsClientAddr = dnsConn.LocalAddr().(*net.UDPAddr)
+		assert.NoError(c, err)
```

Comment:
Should we continue if any of these `assert`'s fire? All of these should return `false` if they fire, so we can return early in that case.

Reply:
yeah that's correct. I think that's what we were doing before, just not handling the race